### ISSUE TITLE
multiline generic definition in <script setup lang="ts" generic=""> cause "Parsing error: ';' expected. eslint".

### ISSUE THREAD

**Issue by pronitsateli731** [vuejs/vue-eslint-parser#217]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I'm using [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue).
- [X] I'm sure the problem is a parser problem. (If you are not sure, search for the issue in [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) repo and open the issue in [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) repo if there is no solution.
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.56.0

### What version of `eslint-plugin-vue` and ` vue-eslint-parser` are you using?

- vue-eslint-parser@9.4.0
- eslint-plugin-vue@9.20.1

### What did you do?

<details>
<summary>Configuration</summary>

In reproduction, I use tsconfig.json and .eslintrc.cjs which were created by "npm init vue@latest".
</details>

```vue
<script setup lang="ts" generic="
  /** definition of data in FormContext */
  TDefinitionOfFormContext extends Record<string, unknown>,
  /** Output of FormContext */
  TOutputOfFormContext,
  /** Vee Validate Path of whole this component */
  TObjectArrayFieldPath extends Path<TDefinitionOfFormContext>,
  /** Row data type */
  TItemValue extends Item<PathValue<TDefinitionOfFormContext, TObjectArrayFieldPath>>
">
  // import and code
</script>
```

### What did you expect to happen?

No error. For example, in .ts file multiline generic definition cause no error, which is indicated in this picture.
![スクリーンショット 2024-01-19 130734](https://github.com/vuejs/vue-eslint-parser/assets/133564027/e3561b8d-020c-4567-9b3e-c6a31c4062ff)

### What actually happened?

Parsing error: ';' expected. eslint
![スクリーンショット 2024-01-19 130803](https://github.com/vuejs/vue-eslint-parser/assets/133564027/c032d769-9866-4a25-8c55-86a88048a081)
![スクリーンショット 2024-01-19 130825](https://github.com/vuejs/vue-eslint-parser/assets/133564027/45a78dcc-f097-4401-87d6-f5c27f7f81a9)

### Link to Minimal Reproducible Example

https://github.com/pronitsateli731/vue-parser-report-r60119

Actually happened:

* src/components/ObjectArrayField.vue
* src/components/PeopleDetailItemRow.vue

Expect to Happen (TypeScript File)

* src/components/useObjectArrayField.ts
